packages/app/src/cli/services/environment.ts

```diff
@@ -225,28 +228,31 @@ async function fetchDevAppAndPrompt(app: AppInterface, token: string): Promise<O
   return reuse ? partnersResponse : undefined
 }
 
-export async function ensureDeployEnvironment(options: DeployEnvironmentOptions): Promise<DeployEnvironmentOutput> {
-  const token = await session.ensureAuthenticatedPartners()
-  let envIdentifiers = await getAppIdentifiers({app: options.app})
+export async function ensureThemeExtensionDevEnvironment(
+  options: DevEnvironmentOptions,
+  extension: ThemeExtension,
+  token: string,
+): Promise<ExtensionRegistration> {
+  const [partnersApp, _] = await fetchAppAndIdentifiers(options, token)
+  const apiKey = partnersApp.apiKey
 
-  let partnersApp: OrganizationApp | undefined
+  const remoteSpecifications = await fetchAppExtensionRegistrations({token, apiKey})
+  const remoteRegistrations = remoteSpecifications.app.extensionRegistrations
 
-  if (options.reset) {
-    envIdentifiers = {app: undefined, extensions: {}}
-  } else if (envIdentifiers.app) {
-    const apiKey = options.apiKey ?? envIdentifiers.app
-    partnersApp = await fetchAppFromApiKey(apiKey, token)
-    if (!partnersApp) throw DeployAppNotFound(apiKey, options.app.packageManager)
-  } else {
-    partnersApp = await fetchDevAppAndPrompt(options.app, token)
+  if (remoteRegistrations.length > 0) {
+    return remoteRegistrations[0]!
   }
```

Comment:
this is not correct, `remoteRegistrations` includes all extensions for the app, of all types. The first one is not necessarily the theme app one. You'd need to filter the list to check if there is a theme app extension and return that.
All this code assumes we won't never increase the allowed number of extensions, but i guess that's an issue to solve in the future :)

Reply:
Great catch! I've applied the fix in a new commit :) - Thank you, @isaacroldan! :)